### PR TITLE
tsp, unbranded, branch for Trait and addQueryParam

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
remove use of `KeyCredentialTrait`

also avoid 3rd param in `RequestOptions.addQueryParam`

### CODE REVIEW

File: javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java [hunk 1]
```diff
@@ -419,10 +416,8 @@ private static void writeQueryParam(MethodParameter parameter, JavaBlock methodB
             };
         } else {
             writeLine = javaBlock -> javaBlock.line(
-                    String.format("requestOptions.addQueryParam(%1$s, %2$s, %3$s);",
-                            ClassType.String.defaultValueExpression(parameter.getSerializedName()),
-                            expressionConvertToString(parameter.getName(), parameter.getClientMethodParameter().getWireType(), parameter.getProxyMethodParameter()),
-                            parameter.getProxyMethodParameter().getAlreadyEncoded()));
+                    getAddQueryParamExpression(parameter,
```
Comment (haolingdong-msft) [inline javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java:419]: I did not see generated code get updated, but from my understanding, this change will impact generated code.

Comment (weidongxu-microsoft) [reply javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java:419]: It is in non-azure code branch, so not affect Azure.

File: javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java [hunk 2]
```diff
@@ -434,6 +429,20 @@ private static void writeQueryParam(MethodParameter parameter, JavaBlock methodB
         }
     }
 
+    private static String getAddQueryParamExpression(MethodParameter parameter, String variable) {
+        // TODO: generic not having 3rd parameter "encoded"
+        if (JavaSettings.getInstance().isBranded()) {
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java:434]: here

File: javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java (hunk 1 above)
Comment (haolingdong-msft) [reply javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java:419]: ok, I see. BTW, may I know why non-azure wants to remove this parameter?

Comment (weidongxu-microsoft) [reply javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java:419]: I don't know. Maybe they do not want to support the case that parameter already encoded.

But anyway they are not finalized, so they could add it back later. But I had to generate code that compiles with generic-core, righ now...

Comment (weidongxu-microsoft) [reply javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java:419]: Same for the Trait. I think they will eventually have Trait.

Comment (haolingdong-msft) [reply javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java:419]: I see, thanks.

**Pull request merged**